File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/library/access/views/ProvideLibraryViews.java
package com.lasthopesoftware.bluewater.client.library.access.views;

import com.lasthopesoftware.bluewater.client.connection.IConnectionProvider;
import com.lasthopesoftware.bluewater.client.library.items.Item;
import com.namehillsoftware.handoff.promises.Promise;

import java.util.List;

public interface ProvideLibraryViews {
	Promise<List<Item>> promiseLibraryViewsFromConnection(IConnectionProvider connectionProvider);
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/library/views/access/ProvideLibraryViews.kt
package com.lasthopesoftware.bluewater.client.browsing.library.views.access

import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId
import com.lasthopesoftware.bluewater.client.browsing.library.views.ViewItem
import com.namehillsoftware.handoff.promises.Promise

fun interface ProvideLibraryViews {
	fun promiseLibraryViews(libraryId: LibraryId): Promise<Collection<ViewItem>>
}
